addons/stock/models/stock_move.py

```diff
@@ -1349,10 +1391,26 @@ def _action_assign(self):
             missing_reserved_quantity = move.product_uom._compute_quantity(missing_reserved_uom_quantity, move.product_id.uom_id, rounding_method='HALF-UP')
             if move._should_bypass_reservation():
                 # create the move line(s) but do not impact quants
-                if move.product_id.tracking == 'serial' and (move.picking_type_id.use_create_lots or move.picking_type_id.use_existing_lots):
+                if move.move_orig_ids:
+                    available_move_lines = _get_available_move_lines(move)
+                    for (location_id, lot_id, package_id, owner_id), quantity in available_move_lines.items():
+                        qty_added = min(missing_reserved_quantity, quantity)
+                        move_line_vals = move._prepare_move_line_vals(qty_added)
+                        move_line_vals.update({
+                            'location_id': location_id.id,
+                            'lot_id': lot_id.id,
+                            'lot_name': lot_id.name,
+                            'owner_id': owner_id.id,
+                        })
+                        move_line_vals_list.append(move_line_vals)
+                        missing_reserved_quantity -= qty_added
+                        if float_is_zero(missing_reserved_quantity, precision_rounding=move.product_id.uom_id.rounding):
+                            break
+
+                if missing_reserved_quantity and move.product_id.tracking == 'serial' and (move.picking_type_id.use_create_lots or move.picking_type_id.use_existing_lots):
                     for i in range(0, int(missing_reserved_quantity)):
                         move_line_vals_list.append(move._prepare_move_line_vals(quantity=1))
-                else:
+                elif missing_reserved_quantity:
```

Comment:
You test missing_reserved_quantity but you add missing_reserved_**uom**_quantity below